src/value/value.rs

```diff
@@ -563,6 +717,27 @@ impl Num {
             Num::F64(v) => Actual::Float(v as f64),
         }
     }
+
+    // /// Given a number, return the narrowest representation of that number.
+    // fn compact(self) -> Self {
```

Comment:
Previously, `to_num_lossy()` used this method to get the lowest fitting integer type. I've now changed it to use `usize` or `isize` irrespective of the narrowest size -- this is commensurate with how values are parsed from environment variables (see `value/parse.rs`). I'm wondering, however, if the compacting behavior was important for applications I can't think of at the moment. And if it is, then we should change both to `to_lossy()` impl _and_ the value parser.

Reply:
If we want to choose the smallest representation, I think that's okay. I don't know if there's any precedence for doing this or not. Deserialization libraries like `json` and `toml` simply choose a representation like `i64` and use it for everything. 

Whatever we do, we should be consistent everywhere. I think the way to go here is to implement `FromStr` for `Num` and have that be the canonical implementation. I'll do that now and compact in it, and we can figure out if that's the most sensible thing or not.